### PR TITLE
Tests: speed up fundrawtransaction test

### PR DESCRIPTION
**Pull request by jnewbery**
Speed up rpc_fundrawtransaction.py

Most of the time in rpc_fundrawtransaction.py is spent waiting for
unconfirmed transactions to propagate. Net processing adds a poisson
random delay to the time it will INV transactions with a mean interval
of 5 seconds. Calls like the following:

```
self.nodes[2].sendrawtransaction(signedTx['hex'])
self.sync_all()
self.nodes[1].generate(1)
````

will therefore introduce a delay waiting for the mempools to sync.
Instead just generate the block on the node that sent the transaction:

```
self.nodes[2].sendrawtransaction(signedTx['hex'])
self.nodes[2].generate(1)
```

rpc_fundrawtransaction.py is not intended to be a test for transaction
relay, so it's ok to do this.

### CODE REVIEW

**Comment by jnewbery** [comment]
h/t to @jonatack for adding logging in #17327. It wouldn't have been obvious where the delays were being introduced without that PR.

**Comment by jnewbery** [comment]
Sample run before this PR:

```
 ./rpc_fundrawtransaction.py 
2019-11-01T14:11:43.898000Z TestFramework (INFO): Initializing test directory /tmp/user/1000/bitcoin_func_test__37epvjg
2019-11-01T14:11:45.270000Z TestFramework (INFO): Connect nodes, set fees, generate blocks, and sync
...
2019-11-01T14:12:41.473000Z TestFramework (INFO): Tests successful
```

(58 seconds)

after:

```
→ ./rpc_fundrawtransaction.py 
2019-11-01T14:20:14.757000Z TestFramework (INFO): Initializing test directory /tmp/user/1000/bitcoin_func_test_sel5qr1p
2019-11-01T14:20:15.609000Z TestFramework (INFO): Connect nodes, set fees, generate blocks, and sync
...
2019-11-01T14:20:27.072000Z TestFramework (INFO): Tests successful
```

(12 seconds)

**Review by jonatack — COMMENTED** [review]
Run times in seconds for me:

ff22751417c6fbbd22f4eefd0e23431a83335c13: 62, 61, 52, 56, 61, 64

a8f7497a2335842dd357fa16ce1b38962a718b2a: 20, 22, 21, 21, 20, 22

**Comment by promag** [comment]
> rpc_fundrawtransaction.py is not intended to be a test for transaction
relay

Concept ACK, nice speedup!

**Comment by jonatack** [comment]
In test_runner.py::BASE_SCRIPTS, rpc_fundrawtransaction.py could be moved down 3 groups to the upper part of `Tests less than 30s`.

**Comment by achow101** [comment]
Are there other tests where we could make a similar improvement?

**Comment by jnewbery** [comment]
> Are there other tests where we could make a similar improvement?

Yes. Anything that is sped up by https://github.com/bitcoin/bitcoin/pull/15881 could be individually sped up by changing how txs are relayed in that test. Alternatively, running the tests under a python profiler and checking which ones spend a lot of time in `sync_mempools` would show places that could be optimized.

**Comment by MarcoFalke** [comment]
If you decide to re-sort the tests, could also include the ones that have been made faster here: https://github.com/bitcoin/bitcoin/issues/16613#issuecomment-541375007

**Comment by jnewbery** [comment]
Made it faster

**Comment by jnewbery** [comment]
> If you decide to re-sort the tests, could also include the ones that have been made faster here: #16613 (comment)

I think that could be done separately

**Comment by MarcoFalke** [comment]
Rebase or force push to work around the GitHub bug?

**Comment by jnewbery** [comment]
> Rebase or force push to work around the GitHub bug?

Done. Thanks!

**Comment by jonatack** [comment]
> Run times in seconds for me:
> 
> [ff22751](https://github.com/bitcoin/bitcoin/commit/ff22751417c6fbbd22f4eefd0e23431a83335c13): 62, 61, 52, 56, 61, 64
> 
> [a8f7497](https://github.com/bitcoin/bitcoin/commit/a8f7497a2335842dd357fa16ce1b38962a718b2a): 20, 22, 21, 21, 20, 22

2919ad318a723c7b90147298d93dd95458c7715f: 20, 21, 22, 24, 21, 21 -> I'm not seeing improvement over a8f7497a2335842dd357fa16ce1b38962a718b2a

File: test/functional/rpc_fundrawtransaction.py [hunk 1]
```diff
@@ -28,6 +28,9 @@ class RawTransactionsTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 4
         self.setup_clean_chain = True
+        # This test isn't testing tx relay. Set whitelist on the peers for
+        # instant tx relay.
+        self.extra_args = [['-whitelist=127.0.01']] * 4
```
Comment (jonatack) [inline test/functional/rpc_fundrawtransaction.py]: s/127.0.01/127.0.0.1/ ?

Credit to @practicalswift caught it my PR ;)

Comment (Sjors) [reply test/functional/rpc_fundrawtransaction.py]: Maybe also make it more specific: `-whitelist=relay@127.0.0.1`.

You could also add `-whitebind` to cover both inbound and outbound, however for some reason this requires you to specify a port, so you'd have to tweak the test suite for that.

**Review by Sjors — APPROVED** [review]
ACK 2919ad3 modulo IP typo (I ran with that corrected).

Tested on macOS 10.15 with `--enable-debug`. Before: 72 seconds. After: 38 seconds.

File: test/functional/rpc_fundrawtransaction.py (hunk 1 above)
Comment (jnewbery) [reply test/functional/rpc_fundrawtransaction.py]: Fixed. Thanks!

**Comment by jnewbery** [comment]
> I'm not seeing improvement over a8f7497

I see at least a second saved on average from not having to stop-start the nodes. I think not stop-starting when unnecessary makes a cleaner test in any case.

File: test/functional/rpc_fundrawtransaction.py [hunk 2]
```diff
@@ -28,6 +28,9 @@ class RawTransactionsTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 4
         self.setup_clean_chain = True
+        # This test isn't testing tx relay. Set whitelist on the peers for
+        # instant tx relay.
+        self.extra_args = [['-whitelist=127.0.0.1']] * 4
```
Comment (jonatack) [inline test/functional/rpc_fundrawtransaction.py]: Could do
```diff
     def set_test_params(self):
-        self.num_nodes = 4
+        num_nodes = 4
+        self.num_nodes = num_nodes
         self.setup_clean_chain = True
         # This test isn't testing tx relay. Set whitelist on the peers for
         # instant tx relay.
-        self.extra_args = [['-whitelist=127.0.0.1']] * 4
+        self.extra_args = [['-whitelist=127.0.0.1']] * num_nodes
```

**Review by jonatack — COMMENTED** [review]
ACK ae6fd926d0b49f69dec4b63e0d5cc2a639408892 happy to re-ACK for the simple change below

File: test/functional/rpc_fundrawtransaction.py (hunk 2 above)
Comment (jnewbery) [reply test/functional/rpc_fundrawtransaction.py]: even better, just:

```
        self.extra_args = [['-whitelist=127.0.0.1']] * self.num_nodes
```

which I've now done.

**Comment by MarcoFalke** [comment]
ACK af7bae734089f6af0029b0887932ccd9a469e12e 🛴

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK af7bae734089f6af0029b0887932ccd9a469e12e 🛴
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUjUQgv/SnHpKGtJxibslQvpMWvhSH/cHWuncQh/jXzz7ygbG1fBHdLK2H68n5tn
RPDqaYnLeK89u85grB2Xf2b8eLJx6Apze9tro7DJAJSnBpMJ7YNuNW1UXoVPJG6W
NwL9v26x/3/Avx1xaxV7ynUziGov+S5adp/yuie6eqSfL3RWX3HGuWOLIYfXU+/d
c9hxXT5/O78DP6Xw2sSFQSDLvcStuNoZ/SgWco3S+Cx93F89XqgECNnDRaeVit9u
ROSVM2p23J63ymFccW5EJngYvHN4LmUcONpUkxBgRrERSDAWN4wG9bBRKFidWrLO
MRaCJGjHHr6zaz8NVSaTqejdHcGCTXQTX7cstq+ssC2ZNBcL7I26+zCg4+OJBNE9
Hb5D7vvLfpP0XKedMGkiB+cvCfcZZOJvr2KALGShiYBL70wctMtsru8ycK9fc8KS
AzCt76h4eGqblgFEw3bPROEG99IqF6HIY12rVlYr2QAqXtKVdraPR4Hx5i+G2xjX
16xuHcdp
=KGsa
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `1ecd30be3396a80fa3c5142b61bd557f16f7db00e6f86414f9df50183d6462cf  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401081ecd30be3396a80fa3c5142b61bd557f16f7db00e6f86414f9df50183d6462cff010cbb9a91e21970287e8c89ff2858bb5d008fff010061a7d0c17744a8d7b82caff199c77ec08f1045dc329f8f008af5660cabf341d660083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff010c4f1874c1728f57c86f79bd2177e7d9008f1045dc329f9f008368cf6e9b3320c6c0083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff010f465c1feeefc8684974dd149ef78da5008f1045dc329f9f00885fb3f4b4fc4c76b0083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010fdbf2ec1528e550191ec0765fbd9b32908f1045dc329f9f0081197d85d202660510083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

**Comment by promag** [comment]
Here we go :owl:

File: test/functional/rpc_fundrawtransaction.py (hunk 2 above)
Comment (jonatack) [reply test/functional/rpc_fundrawtransaction.py]: Yes, even better; doing the same in wallet_avoidreuse.py

**Pull request merged**